xarray/namedarray/pycompat.py

```diff
@@ -89,7 +90,23 @@ def mod_version(mod: ModType) -> Version:
 
 
 def is_chunked_array(x: duckarray[Any, Any]) -> bool:
-    return is_duck_dask_array(x) or (is_duck_array(x) and hasattr(x, "chunks"))
+    return is_duck_dask_array(x) or isinstance(x, _chunkedarray)
+
+
+def has_chunkmanager(x: duckarray[Any, Any]) -> bool:
+    from xarray.namedarray.parallelcompat import get_chunked_array_type
+
+    try:
+        get_chunked_array_type(x)
+    except TypeError as e:
+        if str(e).startswith("Could not find a Chunk Manager which recognises type"):
+            return False
+        elif str(e) == "Expected a chunked array but none were found":
+            return False
+        else:
+            raise  # something else went wrong
```

Comment:
This might be a code smell, in which case `has_chunkmanager`, `guess_chunkmanager`, and `get_chunked_array_type` should be refactored.